storey/drivers.py

```diff
@@ -191,19 +194,31 @@ def _build_feature_store_update_expression(self, aggregation_element, additional
         update_expression = ';'.join(expressions)
         return update_expression, condition_expression, pending_updates
 
+    def _discard_old_pending_items(self, pendings, max_window_millis):
+        res = {}
+        if pendings:
+            last_time = int(list(pendings.keys())[-1] / max_window_millis) * max_window_millis
+            min_time = last_time - max_window_millis
+            for time, value in pendings.items():
+                if time > min_time:
+                    res[time] = value
```

Comment:
Consider deleting in place for performance?

Reply:
i can't delete something i iterating over 🤔